owncloud/fileManagement.js

```diff
@@ -236,12 +280,88 @@ files.prototype.move = function(source, target) {
  */
 files.prototype.copy = function(source, target) {
     return new Promise((resolve, reject) => {
-        helpers._webdavMoveCopy(source, target, 'COPY').then(status => {
-            resolve(status);
+        if (!helpers.getAuthorization()) {
+            reject("Please specify an authorization first.");
+            return;
+        }
+
+        davClient.request('COPY', helpers._buildFullWebDAVPath(source), {
+            'Authorization': helpers.getAuthorization(),
+            'Destination': helpers._buildFullWebDAVPath(target)
+        }).then(result => {
+            if ([200, 201, 204, 207].indexOf(result.status) > -1) {
+                resolve(true);
+            } else {
+                reject(helpers._parseDAVerror(result.body));
+            }
         }).catch(error => {
             reject(error);
         });
     });
 };
 
+files.prototype._parseBody = function(responses) {
+    if (!Array.isArray(responses)) {
+        responses = [responses];
+    }
+    var self = this;
+    var fileInfos = [];
+    for (var i = 0; i < responses.length; i++) {
+        var fileInfo = self._parseFileInfo(responses[i]);
+        if (fileInfo !== null) {
+            fileInfos.push(fileInfo);
+        }
+    }
+    return fileInfos;
+
+};
+
+files.prototype._extractPath = function(path) {
+    var pathSections = path.split('/');
+    pathSections = pathSections.filter(function(section) { return section !== ''});
+
+    let _rootSections = ['remote.php', 'webdav'];
+
+    var i = 0;
+    for (i = 0; i < _rootSections.length; i++) {
+        if (_rootSections[i] !== decodeURIComponent(pathSections[i])) {
+            // mismatch
+            return null;
+        }
+    }
+
+    // build the sub-path from the remaining sections
+    var subPath = '';
+    while (i < pathSections.length) {
+        subPath += '/' + decodeURIComponent(pathSections[i]);
+        i++;
+    }
+    return subPath;
+};
+
+files.prototype._parseFileInfo = function(response) {
+    var path = this._extractPath(response.href);
+    // invalid subpath
+    if (path === null) {
+        return null;
+    }
+    let name = path;
+
+    if (response.propStat.length === 0 || response.propStat[0].status !== 'HTTP/1.1 200 OK') {
+        return null;
+    }
+
+    var props = response.propStat[0].properties;
+    let fileType = 'file';
+    var resType = props['{DAV:}resourcetype'];
+    if (resType) {
+        var xmlvalue = resType[0];
+        if (xmlvalue.namespaceURI === 'DAV:' && xmlvalue.nodeName.split(':')[1] === 'collection') {
```

Comment:
I hope this works in IE 11. should be possible to run the karma tests at least once in a local IE 11 instance once done

Reply:
IE11 is a topic on it's own - afaik we need some polyfills for it anyway .... will create topic ... THX